R/idr0021 - Segmentation.ipynb

```diff
@@ -0,0 +1,677 @@
+    "    }\n",
+    "    rois <- rois[-1,]\n",
+    "    rownames(rois) <- c()\n",
+    "    return(rois)\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rois <- createROIs(features)\n",
+    "rois"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Save the ROIs back to OMERO"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# addROIs creates an ROI for each entry in the dataframe specified by the 'coords' parameter\n",
+    "addROIs(image, coords = rois)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Exercise:** Open the image in the full viewer in OMERO.web and check the ROIs."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Store the features table"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Attach the whole 'features' dataframe to the image:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# 'attachDataframe' directly attaches an R dataframe to an OMERO image (project, dataset, etc.)\n",
+    "# ('invisible' not strictly necessary, just suppresses some unneeded output)\n",
+    "invisible(attachDataframe(image, features, \"ROI features\"))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The dataframe is now attached to the image as an [HDF](https://en.wikipedia.org/wiki/Hierarchical_Data_Format) table with file name 'ROI features'. You can download and open it with software like HDF Compass, load it into python scripts using 'h5py' library, etc. or simply load it directly as R dataframe again."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Alternatively attach it as CSV file:\n",
+    "\n",
+    "- Use R's built-in write.csv method to create a csv file\n",
+    "- And attach that to the image using attachFile function from the romero.gateway"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "csvFile <- \"/tmp/ROI_features.csv\"\n",
+    "write.csv(features, file = csvFile)\n",
+    "fileannotation <- attachFile(image, csvFile)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Automate"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Tasks:\n",
+    "\n",
+    "- Segment each image of a dataset.\n",
+    "- For visualisation create an ROI for each of the objects detected.\n",
+    "- For further analysis only take the largest object per image into account (assumption: this is a centriol), in particular it's area, perimeter and diameter.\n",
```

Comment:
``centriol`` -> ``centriole``   (missing ``e``)
``in particular it's area, perimeter and diameter`` -> ``Parameters to evaluate for this object are area, perimeter and diameter.`` (it is hard to understand what ``it's`` is in the sentence. If there are more parameters than those three, just write ``Main parameters...``